hasjob/views/index.py

```diff
@@ -861,3 +863,51 @@ def manifest():
 @app.route('/api/1/embed.js', methods=['GET'])
 def embed():
     return app.send_static_file('embed.js')
+
+
+@app.route('/api/1/oembed', methods=['GET'], subdomain='<subdomain>')
+@app.route('/api/1/oembed', methods=['GET'])
+@requestargs('url')
+def oembed(url):
+    """
+    Endpoint to support oEmbed - https://oembed.com/
+
+    Example request - http://hasjob.co/api/1/oembed?url=https://hasjob.co/
+
+    Required for services like embed.ly
+    """
+    oembedjs = {}
+
+    parsed = urlparse(url)
+    if not parsed.netloc.endswith(u'hasjob.co'):
+        # is this a legit hasjob domain or subdomain? if not, then return 404
+        abort(404)
+
+    boardname, _ = parsed.netloc.split('.', 1)
+    if boardname == 'hasjob':
+        # in case the url is https://hasjob.co/?embed=1
+        boardname = 'www'
```

Comment:
There's a cleaner way to do this using Flask itself: https://github.com/hasgeek/baseframe/blob/master/baseframe/errors.py#L14-L24

Reply:
I need a way to extract the subdomain from the `url` query param. The linked code helps in recognizing valid url.

Reply:
It will extract the subdomain as well if `SERVER_NAME` is specified in config.

Reply:
We dont want subdomain of current request. Need to extract subdomain from the `url` query param. The `url` query param will contain the page that needs to be embedded. It can be `hasjob.co` or in future, `boardname.hasjob.co`. The `adapter` returned by `bind_to_environment` is valid for the current request. I can't seem to create it from a given url.